src/data/nat/basic.lean

```diff
@@ -372,6 +372,12 @@ begin
   exact lt_succ_iff,
 end
 
+lemma two_lt_of_ne : ∀ {n}, n ≠ 0 → n ≠ 1 → n ≠ 2 → 2 < n
+| 0 h _ _ := (h rfl).elim
+| 1 _ h _ := (h rfl).elim
+| 2 _ _ h := (h rfl).elim
+| (n+3) _ _ _ := (cmp_eq_lt_iff 2 (n + 3)).mp rfl -- this proof is magic to me but it works!
```

Comment:
```suggestion
| (n+3) _ _ _ := (cmp_eq_lt_iff 2 (n + 3)).mp rfl
```

Reply:
I'll remove it, but I am still curious how it works!

Reply:
OK, I've figured out what's going on - this was basically just a wrapper around `dec_trivial`